Support more types

ankon opened the pull request:
Relates to #53

silasbw commented on lib/apps.js:
```diff
@@ -0,0 +1,19 @@
+    ];
+    options = Object.assign({}, options, {
+      path: 'apis/apps',
+      version: options.version || 'v1beta1',
```
How should we handle the petset -> statefulset rename [1]?

The simplest thing would be to add `statefulset` to the `genericTypes` array and add a comment about the rename.  Any better ideas?

[1]: https://github.com/kubernetes/kubernetes/pull/35663

silasbw commented on lib/batch.js:
```diff
@@ -0,0 +1,19 @@
+    ];
+    options = Object.assign({}, options, {
+      path: 'apis/batch',
+      version: options.version || 'v2alpha1',
```
there's a `v1` batch -- should we make that the default?

ankon replied:
Really good question, I was thinking about that as well. 

I guess it makes sense to have the "conservative" default here, anyone that wants alpha/beta stuff should do so knowingly.

silasbw commented on lib/batch.js:
```diff
@@ -0,0 +1,19 @@
+class Batch extends ApiGroup {
+  constructor(options) {
+    const genericTypes = [
+      'scheduledjobs'
```
Similar rename issue as above -- scheduledjobs -> cronjobs [1]

[1]: https://github.com/kubernetes/kubernetes/pull/36021

silasbw commented on lib/common.js:
```diff
@@ -5,8 +5,11 @@ module.exports.aliasResources = function (resourceObject) {
   //   http://kubernetes.io/docs/user-guide/kubectl-overview/
   // and anything else we think is useful.
   const resourceAliases = {
+    clusterroles: [],
```
should we include petsets and scheduledjobs?

silasbw commented on lib/namespaces.js:
```diff
@@ -37,7 +37,10 @@ class Namespaces extends BaseObject {
     // Generic objects we don't implement special functionality for.
     //
     const genericTypes = [
+      'clusterroles',
```
should we include petsets and scheduledjobs?

silasbw commented on lib/api.js:
```diff
@@ -2,9 +2,15 @@
 const Extensions = require('./extensions');
+const Apps = require('./apps');
+const Batch = require('./batch');
+const RBAC = require('./rbac');
```
Our convention has been to avoid consecutive caps (e.g., `apiUrl`, not `apiURL`).  So this would be `Rbac` not `RBAC`.  `Rbac` looks funny, I know, but it follows the convention we're currently using.

ankon commented:
I've fixed the nits above and pushed for a quick look at it - but tests are still missing and coming.

This is now rebased this onto #59, while I'm working on the tests :)

silasbw commented:
Did you make progress on tests?  or get waylaid by that cleanup that needed to happen :)

Is there something I can do to help you finish this?

silasbw commented:
Rebased/updated branch here: https://github.com/godaddy/kubernetes-client/pull/78

Thanks again @ankon for initiating these changes.
